save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -102,5 +94,44 @@ internal fun String.extractWarning(warningRegex: Regex,
         throw ResourceFormatException("Could not extract file name from line [$this], cause: ${e.message}")
     }
 
-    return extractWarning(warningRegex, fileName, lineGroupIdx, columnGroupIdx, messageGroupIdx)
+    return extractWarning(false, warningRegex, fileName, lineGroupIdx, columnGroupIdx, messageGroupIdx)
+}
+
+/**
+ * @param defaultLineMode parameter for default line
+ * @param warningRegex regular expression for warning
+ * @param fileName file name
+ * @param lineGroupIdx index of capture group for line number or line number
+ * @param columnGroupIdx index of capture group for column number
+ * @param messageGroupIdx index of capture group for waring text
+ * @return a [Warning] or null if [this] string doesn't match [warningRegex]
+ * @throws ResourceFormatException when parsing a file
+ */
+@Suppress(
+    "TOO_MANY_PARAMETERS",
+    "LongParameterList",
+    "TooGenericExceptionCaught",
+    "SwallowedException")
+internal fun String.extractWarning(defaultLineMode: Boolean,
```

Comment:
I'd rather add a new method `getLineNumber(defaultLineMode, lineGroupIdx, defaultLineNumber)` and call it from `mapNotNull` in `WarnPlugin`. And later this method can be enhanced with placeholders logic